src/cpp/src/tokenizer.cpp

```diff
@@ -109,17 +78,41 @@ class Tokenizer::TokenizerImpl {
 
     std::string m_chat_template = {};
 
+    std::pair<int, int> get_padding_values(PaddingMode padding_mode, size_t max_length) {
+        switch (padding_mode) {
+            case PaddingMode::TRUNCATE:
+                return {max_length, 0};
+            case PaddingMode::LONGEST:
+                return {std::numeric_limits<int32_t>::max(), 0};
+            case PaddingMode::MAX_LENGTH:
+                return {max_length, max_length};
+            default:
+                OPENVINO_THROW("Unknown padding mode");
+        }
+    }
+
     void set_state_if_necessary(CircularBufferQueueElementGuard<ov::InferRequest>& infer_request_guard, const ov::AnyMap& params) {
         bool add_special_tokens_flag = m_add_special_tokens;
         bool skip_special_tokens_flag = m_skip_special_tokens;
-
+        std::optional<size_t> max_length_val;
+        PaddingMode padding_mode_val = PaddingMode::TRUNCATE;
+        
         ov::genai::utils::read_anymap_param(params, add_special_tokens.name(), add_special_tokens_flag);
         ov::genai::utils::read_anymap_param(params, skip_special_tokens.name(), skip_special_tokens_flag);
-
-        // If user requested add_special_tokens mode different from the current one,
-        // need to set state variable.
-        // If requested mode matches the stored state set, then don't touch states.
-        if (add_special_tokens_flag == m_add_special_tokens && skip_special_tokens_flag == m_skip_special_tokens) {
+        ov::genai::utils::read_anymap_param(params, padding_mode.name(), padding_mode_val);
+        ov::genai::utils::read_anymap_param(params, max_length.name(), max_length_val);
+        
+        int max_trunc_length_val = m_max_trunc_length;
+        int max_pad_length_val = m_max_pad_length;
+
+        std::tie(max_trunc_length_val, max_pad_length_val) = get_padding_values(padding_mode_val, *max_length_val);
```

Comment:
MAX_LENGTH and missing max_length_val is  undefined behavior.

Reply:
I had a solution for this. But since approach changed a bit, this code is not present anymore.